fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/util/ParameterHashUtilTest.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.jdbc.test.util;
+
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.util.Arrays;
+
+import org.testng.annotations.Test;
+
+import com.ibm.fhir.persistence.jdbc.dto.CompositeParmVal;
+import com.ibm.fhir.persistence.jdbc.dto.DateParmVal;
+import com.ibm.fhir.persistence.jdbc.dto.LocationParmVal;
+import com.ibm.fhir.persistence.jdbc.dto.NumberParmVal;
+import com.ibm.fhir.persistence.jdbc.dto.QuantityParmVal;
+import com.ibm.fhir.persistence.jdbc.dto.ReferenceParmVal;
+import com.ibm.fhir.persistence.jdbc.dto.StringParmVal;
+import com.ibm.fhir.persistence.jdbc.dto.TokenParmVal;
+import com.ibm.fhir.persistence.jdbc.util.ExtractedSearchParameters;
+import com.ibm.fhir.persistence.jdbc.util.ParameterHashUtil;
+import com.ibm.fhir.persistence.jdbc.util.type.NumberParmBehaviorUtil;
+import com.ibm.fhir.search.date.DateTimeHandler;
+import com.ibm.fhir.search.util.ReferenceValue;
+import com.ibm.fhir.search.util.ReferenceValue.ReferenceType;
+
+/**
+ * Utility to do testing of the parameter hash utility.
+ */
+public class ParameterHashUtilTest {
```

Comment:
Need to add a test case where a pair of values is swapped between two parameters.

param1: value1
param2: value2

compare with:

param1: value2
param2: value1

The computed hash should be different (but with the current implementation I believe the hashes will come out the same).

Reply:
I added testcase testExtractedParametersSwappedValues(), which shows that the values do come out different, since the parameter (name/code) is part of the hash.

Reply:
Testcase updated after sorting and using a visitor to generate the hash.